lib/plenty_client/request.rb

```diff
@@ -5,22 +5,16 @@ module PlentyClient
   module Request
     module ClassMethods
       def request(http_method, path, params = {})
-        return false if http_method.nil? || path.nil?
-        return false unless %w[post put patch delete get].include?(http_method.to_s)
+        raise ArgumentError, "http_method or path is missing" if http_method.nil? || path.nil?
+        unless %w[post put patch delete get].include?(http_method.to_s)
+          raise ArgumentError, "unsupported http_method: #{http_method}"
+        end
 
         login_check unless PlentyClient::Config.tokens_valid?
 
         params = stringify_symbol_keys(params) if params.is_a?(Hash)
 
-        attempts = PlentyClient::Config.attempt_count
-        attempts.times do
-          begin
-            response = perform(http_method, path, params)
-            return response if response
-          rescue Faraday::ConnectionFailed => e
-          end
-        end
-        raise PlentyClient::AttemptsExceeded, "unable to get valid response after #{attempts} attempts"
+        perform(http_method, path, params)
```

Comment:
This retry routine was generally very helpful as Plenty is prone to just have hickups at times, we should try to keep it while still maintaining you newly introduced fine grained error reporting.